android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -27,17 +27,23 @@ open class FullScreenFeature(
     private val fullScreenChanged: (Boolean) -> Unit,
 ) : LifecycleAwareFeature, UserInteractionHandler {
     private var scope: CoroutineScope? = null
+    private var isFullscreenCustomTab: Boolean = false
     private var observation: Observation = createDefaultObservation()
 
     /**
      * Starts the feature and a observer to listen for fullscreen changes.
      */
     override fun start() {
-        scope = store.flowScoped { flow ->
-            flow.map { state -> state.findTabOrCustomTabOrSelectedTab(tabId) }
-                .map { tab -> tab.toObservation() }
-                .ifChanged()
-                .collect { observation -> onChange(observation) }
+        if (tabId != null && store.state.findCustomTab(tabId)?.content?.fullScreen == true) {
+            isFullscreenCustomTab = true
+            fullScreenChanged(true)
+        } else {
```

Comment:
Why do we need this? If we are sending a CustomTabState with fullscreen enabled, (what is currently in the else) should be enough to handle this case.

Reply:
If we remove the if-check, everything still works as expected:

```diff
diff --git a/android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt b/android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt
index 536a31043a..34f85aa4bf 100644
--- a/android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt
+++ b/android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt
@@ -34,17 +34,17 @@ open class FullScreenFeature(
      * Starts the feature and a observer to listen for fullscreen changes.
      */
     override fun start() {
-        if (tabId != null && store.state.findCustomTab(tabId)?.content?.fullScreen == true) {
-            isFullscreenCustomTab = true
-            fullScreenChanged(true)
-        } else {
+//        if (tabId != null && store.state.findCustomTab(tabId)?.content?.fullScreen == true) {
+//            isFullscreenCustomTab = true
+//            fullScreenChanged(true)
+//        } else {
             scope = store.flowScoped { flow ->
                 flow.map { state -> state.findTabOrCustomTabOrSelectedTab(tabId) }
                     .map { tab -> tab.toObservation() }
                     .ifChanged()
                     .collect { observation -> onChange(observation) }
             }
-        }
+//        }
     }

     override fun stop() {

```